### PR TITLE
add doublePow to primitives

### PR DESCRIPTION
**Pull request by AliasQli**
This PR adds `prim__doublePow`. This solves [issue #1432](https://github.com/idris-lang/Idris2/issues/1432).

### CODE REVIEW

**Comment by stefan-hoeck** [comment]
Thanks a lot for this! Please note the following: Right now, continuous integration tests on github will verify, if the libs are buildable by the bootstrap compiler, because the bootstrap compiler doesn't have its own set of standard libraries. Since bootstrap doesn't know about the new `prim__doublePow` primitive yet, we have to wait with adding this to the `prelude` until the next minor release (when bootstrap will be updated) or until #1391, which deals with exactly this issue, is merged.

@gallais : There hasn't been a lot of feedback on #1391. Do you know what the status there is?

One minor thing: If you change stuff in `Core.TTC`, you also need to increase `Core.Binary.ttcVersion`.

**Comment by AliasQli** [comment]
Sorry, I misclosed the PR. Reopened again.

File: libs/prelude/Prelude/Types.idr [hunk 1]
```diff
@@ -699,7 +699,7 @@ log x = prim__doubleLog x
 
 public export
 pow : Double -> Double -> Double
-pow x y = exp (y * log x)
+pow x y = prim__doublePow x y -- exp (y * log x)
```
Comment (gallais) [inline libs/prelude/Prelude/Types.idr:702]:
```suggestion
pow x y = exp (y * log x) -- prim__doublePow x y
```

File: src/Core/Binary.idr [hunk 2]
```diff
@@ -33,7 +33,7 @@ import Data.Buffer
 -- TTC files can only be compatible if the version number is the same
 export
 ttcVersion : Int
-ttcVersion = 54
+ttcVersion = 55
```
Comment (gallais) [inline src/Core/Binary.idr:36]:
```suggestion
ttcVersion = 56
```

File: src/Core/Hash.idr [hunk 3]
```diff
@@ -350,31 +350,33 @@ Hashable (PrimFn arity) where
...
 
     Cast f t =>
-      h `hashWithSalt` 35 `hashWithSalt` f `hashWithSalt` t
+      h `hashWithSalt` 36 `hashWithSalt` f `hashWithSalt` t
     BelieveMe =>
-      h `hashWithSalt` 36
-    Crash =>
       h `hashWithSalt` 37
+    Crash =>
+      h `hashWithSalt` 38
```
Comment (gallais) [inline src/Core/Hash.idr:379]: Can we just put `DoublePow` at the end and not shift everything by one?

**Comment by AliasQli** [comment]
Still can't bootstrap via racket.

**Comment by gallais** [comment]
Please don't update the bootstrap, there is no good reason to do so AFAICT

**Comment by AliasQli** [comment]
Ok, I understand. But if so, I'll have to change it back to

```idris
pow x y = exp (y * log x)
```

Then why should I add `prim__doublePow`? Or is there another way to work around it?

**Comment by gallais** [comment]
> But if so, I'll have to change it back to

This is indeed what I suggested in the review.

> Then why should I add `prim__doublePow`?

We are about to release a new version & the definition of `pow` can be updated once
the version with the new primitive has been released.

**Comment by AliasQli** [comment]
Okay. Now those changes have been reverted.

**Comment by gallais** [comment]
> Now those changes have been reverted.

You still have the updated bootstrap

**Comment by edwinb** [comment]
I've resolved the conflicts and made a few other changes. I'll submit a new PR to check everything is okay with the latest version and close this one. Note that since inlining happens, this does need two rounds of build and install to actually get the right result from `pow`, but that's to be expected.

**Comment by edwinb** [comment]
Oh, right, it'll also need the prelude updated and a wait for the next release. Adding new primitives can be awkward like that.

**Pull request closed without merging**